test/python/providers/test_fake_backends.py

```diff
@@ -0,0 +1,63 @@
+# -*- coding: utf-8 -*-
+
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2020.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+# pylint: disable=missing-class-docstring,missing-function-docstring
+# pylint: disable=missing-module-docstring
+
+import operator
+
+from test import combine
+from ddt import ddt
+
+from qiskit.circuit import QuantumCircuit
+from qiskit.execute import execute
+from qiskit.test.base import QiskitTestCase
+from qiskit.test.mock import backends
+from qiskit.test.mock.fake_backend import HAS_AER
+
+
+FAKE_BACKENDS = [
+    x for x in dir(backends) if x.startswith('Fake')]
```

Comment:
Could you access these through the `FakeProvider`?

Reply:
That's a good call, I forgot that was there. I'll update it right now to use the provider instead of just searching the directory.

Reply:
Ok, that was a bit more involved than I was expecting. It caught some additionall issues in the fake backends. Done in: https://github.com/Qiskit/qiskit-terra/pull/4018/commits/bcd8a6e3a5da90bb70e5fb8d30a5fd0604099077